src/compiler/compile.js

```diff
@@ -0,0 +1,28 @@
+import * as compiler from '@ampproject/bento-compiler';
+
+import {getBuilders} from './builders';
+
+/**
+ * Returns the AST for an AMP Document with eligible components server-rendered.
+ *
+ * @param {import('./types').CompilerRequest} request
+ * @return {import('./types').CompilerResponse}
+ */
+export function compile(request) {
+  if (!request || (!request.document && !request.nodes)) {
+    throw new Error('Must provide either document or nodes');
+  }
+
+  // TODO(samouri): remove the defaults.
+  const versions = request.versions ?? [
+    {component: 'amp-layout', version: 'v0'},
+  ];
+
+  const {document, nodes} = request;
```

Comment:
nit: may as well do this as the first line in the function
```javascript
const {document, nodes, version} = request || {};
if (!versions || !(document || nodes)) {
  throw ...
}
```

Reply:
It may be worth removing the JS default now since it's easier/faster to include it on the c++ side, whereas this needs to roll out more slowly with our AMP releases